### DISCUSSION TITLE
How to remove stray voxels after converting mesh

### DISCUSSION THREAD

**Discussion by anthops** [pyvista/pyvista#1746] [Q&A]
Hi. I'm currently experimenting with pv.voxelize() to turn meshes into voxel format.
However, when running my code, I've noticed that there are "outlier" voxels appearing in unexpected places. 
For example, here is the voxelization of pv.Sphere():

![image](https://user-images.githubusercontent.com/32993852/138565768-fdce2060-d6a3-4b5d-a0fc-371ed9231e66.png)

As you can see, there are 2 stray voxels outside of the sphere.
It gets worse when I use more complex shapes, such as examples.download_cow(): 

![image](https://user-images.githubusercontent.com/32993852/138565788-8708d472-5b6a-46d6-b279-9cad5ce536cf.png)

Here is the code I am using for each. Where for the sphere I just use mesh = pv.Sphere(). Please let me know if I'm doing anything wrong or have missed something :)

Thank you

```py
import numpy as np

import pyvista as pv
from pyvista import examples

mesh = examples.download_cow()

density = mesh.length/100
voxels = pv.voxelize(mesh, density)

plotter = pv.Plotter()

plotter.add_mesh(voxels, opacity = 0.5)

plotter.show()

```

**Comment by banesullivan** [comment]
I'm not seeing any outlier voxels when doing this on the Sphere or the cow...

```py
import pyvista as pv
from pyvista import examples

mesh = examples.download_cow()

density = mesh.length/100
voxels = pv.voxelize(mesh, density)

plotter = pv.Plotter()

plotter.add_mesh(voxels,)

plotter.show()
```
![Screen Shot 2021-11-19 at 10 33 28 AM](https://user-images.githubusercontent.com/22067021/142666197-daafa7ab-2333-4b80-85fd-ea2e698d0b0d.png)

With that said, you could remove outliers by using the `connectivity()` filter. Here is an example: https://docs.pyvista.org/examples/01-filter/connectivity.html
W
